scripts/ci-entrypoint.sh

```diff
@@ -108,17 +112,17 @@ wait_for_nodes() {
 
     # Ensure that all nodes are registered with the API server before checking for readiness
     local total_nodes="$((CONTROL_PLANE_MACHINE_COUNT + WORKER_MACHINE_COUNT))"
-    while [[ $(kubectl get nodes -ojson | jq '.items | length') -ne "${total_nodes}" ]]; do
+    while [[ $(${KUBECTL} get nodes -ojson | jq '.items | length') -ne "${total_nodes}" ]]; do
```

Comment:
Have you tested with with `ci-entrypoint.sh`? I am not sure this is going to work becuase `${KUBECTL}` points to `hack/tools/bin/kubectl` which is a folder.

Reply:
I ran `make verify`, it runs all the scripts as well right? or am I missing something😶?

Reply:
`make verify` runs some checks to ensure files are upto date (has correct headers, go mod is upto date, go generate was run, linting of shell scripts) but doesn't run e2e scripts or other processes:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/4fe1b243d2c694a70b8a154ca9ae13f7d68a5149/Makefile#L603-L624

Reply:
I will test `ci-entrypoint.sh` once I understand is it supposed to be `mkdir -p hack/tools/bin/kubectl` or `mkdir -p hack/tools/bin`. The first one gives `folder already exists` and the latter `realpath:......: No such file or directory`.